elfloader-tool/src/common.c

```diff
@@ -618,3 +623,48 @@ WEAK void platform_init(void)
 {
     /* nothing by default */
 }
+
+#ifdef CONFIG_ARCH_ARM
+extern void flush_dcache(void);
+extern void invalidate_icache(void);
+#endif
+
+/*
+ * Moves the elfloader to its expected start address in memory.
+ * This is for when the elfloader needs to be in a specific location
+ * to ensure that it doesn't occupy memory that it needs to load the
+ * kernel and user applications into.
+ */
+char * fixup_image_base(void) {
+    char *load_base = _start;
```

Comment:
This will only work if this assignment uses a pc relative address. This code isn't loaded compiled with any position independent flags. So this might be a bit too dodgy.  Maybe it should take the real base address as an argument.
```
    80a1c752:   ffffe797                auipc   a5,0xffffe
    80a1c756:   8ae78793                add     a5,a5,-1874 # 80a1a000 <_start>
```